specreduce/core.py

```diff
@@ -2,3 +2,246 @@
 # This module implements the base CCDPROC functions
 from __future__ import (absolute_import, division, print_function,
                         unicode_literals)
+import os
+# import re
+# from copy import deepcopy
+# from datetime import datetime
+# from importlib import import_module
+# import warnings
+
+import matplotlib.pyplot as plt
+# import matplotlib.cm as cm
+# from matplotlib.widgets import Cursor
+
+# from astropy.convolution import convolve, Box1DKernel
+# from astropy import units as u
+# from astropy.io import fits
+# from astropy.modeling import models, fitting
+# from astropy.stats import sigma_clip
+# from astropy.table import Table
+# from astropy.wcs import WCS
+
+import numpy as np
+
+# from scipy.optimize import curve_fit
+# import scipy.signal
+from scipy.interpolate import UnivariateSpline
+# from scipy.interpolate import BSpline, SmoothBivariateSpline
+
+# from specutils import SpectralRegion
+
+__all__ = ["genfluxcal", "applyfluxcal"]
+
+
+def _mag2flux(wave, mag, zeropoint=48.60):
+    """
+    An internal method to convert quantities in magnitudes to flux units.
+    Important for handling standard stars as well as other data files ported
+    from IRAF, such as the data files in the specreduce/datasets/onedstds
+    directory. Basically converts to "PHOTFLAM" units. Assumes that
+    the common flux zero point as used in IRAF.
+
+    Parameters
+    ----------
+    wave : a 1D numpy array
+        The wavelength of the data points.
+
+    mag : a 1D numpy array
+        The magnitudes of the data points.
+
+    zeropoint : float, optional
+        Default is set to 48.60. Conversion factor for converting mag into flux.
+
+    Returns
+    -------
+    Corresponding quantity in flux units.
+    """
+
+    c = 2.99792458e18  # speed of light, in Angstrom/second
+    flux = 10.0 ** ((mag + zeropoint) / (-2.5))
+    return flux * (c / wave ** 2.0)
+
+
+def genfluxcal(obj_wave, obj_flux, std, mode="spline", polyorder=9, show=False):
+    """
+    Generate the sensitivity function (or sensfunc) needed for the flux calibration of
+    a 1D spectrum.
+
+    Parameters
+    ----------
+    obj_wave : a 1D numpy array
+        The 1D wavelength array of the spectrum to be flux calibrated.
+
+    obj_flux : a 1D numpy array
+        The 1D flux array of the spectrum to be flux calibrated.
```

Comment:
Something to discuss, but instead of an wave and flux here, I think we want to have users input a Spectrum1D object to the function.

Reply:
Yeah, true

Reply:
I agree, but can (should) we support _both_ an array or a Spectrum1D object?

Reply:
Yeah, we could support both a `numpy` array and `Spectrum1D` object by checking the type with `isinstance()` first. But personally would prefer using `Spectrum1D`.

Reply:
If it's an array, you need both a wavelength and a flux.  In the future, you might also need the error array and data quality array or other information.  Either way, this will be additional parameters in the function call (which can be confusing and harder to maintain) and then checking of those parameters (which can be more to maintain and redundant work).   

For these reasons, I would argue passing in the Astropy objections to the functions would be a best practice unless there is another python native way to represent it or no other information is needed.